sched/scheduler/cluster_state.go

```diff
@@ -107,30 +107,21 @@ func newNodeState(node cluster.Node) *nodeState {
 // nodes added or removed based on the supplied channel. ReadyFn is optional.
 // New cluster is returned along with a doneCh which the caller can close to exit our goroutine.
 func newClusterState(initial []cluster.Node, updateCh chan []cluster.NodeUpdate, rfn ReadyFn) *clusterState {
-	nodes := make(map[cluster.NodeId]*nodeState)
-	nodeGroups := map[string]*nodeGroup{"": newNodeGroup()}
-	suspendedNodes := map[cluster.NodeId]*nodeState{}
+	var updates []cluster.NodeUpdate
 	for _, n := range initial {
-		nodes[n.Id()] = newNodeState(n)
-		nodeGroups[""].idle[n.Id()] = nodes[n.Id()]
-		if rfn == nil {
-			close(nodes[n.Id()].readyCh)
-		} else {
-			nodes[n.Id()].startReadyLoop(rfn)
-		}
-	}
-	if rfn != nil {
-		nodes, suspendedNodes = suspendedNodes, nodes
+		updates = append(updates, cluster.NewAdd(n))
 	}
-	return &clusterState{
+	cs := &clusterState{
 		updateCh:         updateCh,
-		nodes:            nodes,
-		suspendedNodes:   suspendedNodes,
-		nodeGroups:       nodeGroups,
+		nodes:            make(map[cluster.NodeId]*nodeState),
+		suspendedNodes:   map[cluster.NodeId]*nodeState{},
+		nodeGroups:       map[string]*nodeGroup{"": newNodeGroup()},
 		maxLostDuration:  defaultMaxLostDuration,
 		maxFlakyDuration: defaultMaxFlakyDuration,
 		readyFn:          rfn,
 	}
+	cs.update(updates)
```

Comment:
This was entirely duplicate code that was already present in cs.update(). That said, I committed this before deciding to make a new scheduler related PR.